test/e2e/performanceprofile/functests/utils/node_inspector/inspector.go

```diff
@@ -0,0 +1,191 @@
+		return err
+	}
+	return nil
+}
+
+func IsRunning(cli client.Client, namespace, name string) (bool,error){
+	return daemonset.IsRunning(cli, namespace, name)
+}
+
+func createDaemonSet(name, namespace, serviceAccountName, image string) *appsv1.DaemonSet {
+	MountPropagationHostToContainer := corev1.MountPropagationHostToContainer
+	return &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"name": name,
+			},
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": name,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"target.workload.openshift.io/management": `{"effect": "PreferredDuringScheduling"}`,
+					},
+					Labels: map[string]string{
+						"name": name,
+					},
+				},
+				Spec: corev1.PodSpec{
+					HostPID:                       true,
+					HostNetwork:                   true,
+					ServiceAccountName:            serviceAccountName,
+					TerminationGracePeriodSeconds: pointer.Int64(0),
+					NodeSelector:                  map[string]string{"kubernetes.io/os": "linux"},
+					Containers: []corev1.Container{
+						{
+							Name:            "node-daemon",
+							Image:           image,
+							Command:         []string{"/bin/bash", "-c", "sleep INF"},
+							ImagePullPolicy: corev1.PullAlways,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									"cpu":    resource.MustParse("20m"),
+									"memory": resource.MustParse("50Mi"),
+								},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								Privileged:             pointer.Bool(true),
+								ReadOnlyRootFilesystem: pointer.Bool(true),
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									MountPath:        "/rootfs",
+									Name:             "rootfs",
+									MountPropagation: &MountPropagationHostToContainer,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "rootfs",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+								},
+							},
+						},
+					},
+					Tolerations: []corev1.Toleration{
+						{
+							Operator: corev1.TolerationOpExists,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createServiceAccount(name, namespace string) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+}
+
+func createClusterRole(name string) *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"nodes"},
+				Verbs:     []string{"get", "list", "watch"},
```

Comment:
We discuss about it offline, are these needed?

Reply:
Checked that now on a certain unit test, seems like it is redundant indeed.

Reply:
Please also check the privileged permissions, since we're only reading, i'm no sure it's needed

Reply:
I don't think we are only reading from the filesystem, we are executing commands.

Reply:
for example here:
https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go#L271